spec/fixtures/pyproject.toml

```diff
@@ -7,6 +7,13 @@ authors = ["Tyrel Souza <tyrel@tidelift.com>"]
 [tool.poetry.dependencies]
 python = "^3.7"
 django = "^3.0.7"
+pathlib2 = [
+  { version = "2.3.7.post1", python = ">=3.7" },
+  { version = "2.3.6", markers = "python_version < '3.7' and sys_platform == 'win32'" },
+  { version = "2.3.5", python = "<3.7", platform = "linux" },
+  # same version but from different source -- SHALL be treated as non-identical
+  { git = "https://github.com/jazzband/pathlib2.git", tag = "2.3.5", python = "<3.7", platform = "darwin" },
+]
```

Comment:
Here's the case this PR is fixing. We'll throw away all the "environment markers" in each version specifier, but we could theoretically add those to a new Dependency field in the future, or append them to "requirement".